distribution/openrct2.d.ts

```diff
@@ -1246,7 +1245,65 @@ declare global {
         readonly elements: TrackSegmentElement[];
 
         /**
-         * Gets a length of the subpositions list for this track segment.
+         * The curve direction of the suggested following piece, or if the suggested following piece
+         * is a track segment
+         */
+        readonly nextCurve: TrackCurveType;
+
+        /**
+         * The curve direction of the suggested preceding piece, or if the suggested preceding piece
+         * is a track segment
+         */
+        readonly previousCurve: TrackCurveType;
+
+        /**
+         * The track segment for the following piece. Only valid if chainEndCurve is trackSegment.
+         */
+        readonly nextElement: number | null;
+
+        /**
+         * The track segment for the preceding piece. Only valid if chainBeginCurve is trackSegment.
+         */
+        readonly previousElement: number | null;
+
+        /**
+         * The base price of the track segment.
+         */
+        readonly priceModifier: number;
+
+        /**
+         * Track segment representing the mirror image of the track segment.
+         */
+        readonly mirrorElement: number | null;
+
+        /**
+         * Track segment representing the covered/flume variant of the track segment.
+         */
+        readonly alternateType: number | null;
+
+        /**
+         * Which direction the track curves towards.
+         */
+        readonly turnDirection: TrackCurveType;
+
+        /**
+         * Which direction the track slopes towards.
+         */
+        readonly slopeDirection: TrackSlopeType;
+
+        readonly onlyAllowedUnderwater: boolean;
+        readonly onlyAllowedAboveGround: boolean;
+        readonly allowsChainLift: boolean;
+        readonly isBanked: boolean;
+        readonly isGolfHole: boolean;
+        readonly isHelix: boolean;
+        readonly startsHalfHeightUp: boolean;
+        readonly isSteepUp: boolean;
```

Comment:
is there another accessor for isSteepDown? An alternate exposure could be "isSteep" + TrackCurveType.Down

Reply:
Same line of thinking, how do vertical track pieces get exposed in this schema?

Reply:
There are no accompanying values for isSteepUp, nor is there anything for any other slope angle. This flag is present for the sole purpose of limiting steep chain lifts to reverse sections for reverse incline shuttle mode but disallowing steep chain lifts on forward pieces.

Reply:
that's a great comment to add. thanks.